src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] storage;
+    private int capacity;
+    private int size;
+    private int threshold;
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private final Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity == 0) {
+            storage = new Node[DEFAULT_CAPACITY];
+            capacity = DEFAULT_CAPACITY;
+            threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        }
+        resizeIfNeed();
+        if (addNode(key, value)) {
+            ++size;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (storage == null) {
+            return null;
+        }
+        return findNode(key) == null ? null : findNode(key).value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        return key == null ? 0 : key.hashCode();
+    }
+
+    private int getIndex(K key) {
+        return Math.abs(getHash(key)) % capacity;
+    }
+
+    private Node<K, V> findNode(K key) {
+        Node<K, V> node = storage[getIndex(key)];
+        if (node != null) {
+            K nodeKey;
+            do {
+                nodeKey = node.key;
+                if ((nodeKey == key || nodeKey != null && nodeKey.equals(key))) {
```

Comment:
from this task on, you can use `Objects.equals()`
it will make your code cleaner